### PR TITLE
Add Cloud guide

### PR DESCRIPTION
**Pull request by raddevon**

### CODE REVIEW

**Comment by raddevon** [comment]
One thing I want to rope into this: I want to write alt text for all the images. There's something wrong with that right now, so @jaclarke is helping me out with it. Wanted to go ahead and get this PR in just in case it needs to be merged before that's done.

Oh, and you may not love what I've done to make the "Join Cloud Waitlist" button work here. I just applied some inline styles. Dropping the raw HTML in didn't work. It stretched the full width of the text column.

File: docs/guides/cloud.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+.. warning::
+
+    EdgeDB Cloud is currently in beta. This guide will not work unless you have
+    access. To request access, click the "Join Cloud Waitlist" button below or
+    join `our Discord <https://discord.gg/umUueND6ag>`_ and request access
+    there.
+
+.. raw:: html
+
+    <a class="pageNav_promoLink__OSucb" href="/p/cloud-waitlist"
```
[the remark this answers is not in the document]
Comment (raddevon) [reply docs/guides/cloud.rst:18]: Makes sense, @jaclarke. I'm open to either. Maybe a global class is simpler?

File: docs/guides/cloud.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+===================
+
+To use Cloud via the CLI, first log in using :ref:`ref_cli_edgedb_cloud_login`.
+Once your login is successful, you will be able to create an instance using
+either :ref:`ref_cli_edgedb_instance_create` or
+:ref:`ref_cli_edgedb_project_init`, depending on whether you also want to
+create a local project linked to your instance.
+
+In either case, you will pass an instance name matching the EdgeDB Cloud
+instance name format: ``<github-username>/<instance-name>``
```
Comment (zackelan) [inline docs/guides/cloud.rst:37]: we should document the instance name requirements somewhere, see https://github.com/edgedb/shared-client-testcases/pull/27 for the canonical list

File: docs/guides/cloud.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (raddevon) [reply docs/guides/cloud.rst:18]: Actually, I don't think the button makes sense in this context. For one, there's another copy of the same button just in the navigation bar which is really close to this one. Two, the warning paragraph above it talks about the button and tells you to click it below. I feel like it makes more sense to just have a clickable link inline in that warning paragraph instead. I think @1st1 wanted the button here, but seeing it in place, I'm not sure it works that well in this context.

Maybe we would still want to be able to place those buttons in other places, so the component or global class might be useful, but I don't think they're necessary for this guide.

File: docs/guides/cloud.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+.. raw:: html
+
+    <a class="pageNav_promoLink__OSucb" href="/p/cloud-waitlist"
+      style="width:152px;margin-left:0;text-align:center;">
+      <span>Join Cloud Waitlist</span>
+    </a>
+
+EdgeDB Cloud is the easiest way to host your EdgeDB instance. You have two
+options for interfacing with EdgeDB Cloud: via our CLI or through a graphical
```
[the remark this answers is not in the document]
Comment (raddevon) [reply docs/guides/cloud.rst]: Good call! Changed this to "We offer two ways to interact with EdgeDB Cloud: via our CLI or through a graphical web interface."

[the remark this answers is not in the document]
Comment (1st1) [reply docs/guides/cloud.rst]: Maybe

> You can use EdgeDB Cloud via our `edgedb` command-line tool and/or through a web interface at https://cloud.edgedb.com.

File: docs/guides/cloud.rst (hunk 2 above)
Comment (1st1) [reply docs/guides/cloud.rst:29]: * instance names cannot start with a dash `-` or contain double dashes `--`
* only latin alpha-numeric characters, `_`, or `-`
* maximum instance name length is 61 characters minus the length of your GitHub username

Comment (1st1) [reply docs/guides/cloud.rst:29]: Devon, we're going to be rapidly iterating the cloud UI. Let's drop the screenshots entirely.

File: docs/guides/cloud.rst [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+``<github-username>/<instance-name>`` format when prompted interactively.
+
+
+Web Cloud Interface
+===================
+
+If you'd prefer, you can also manage your account via `the EdgeDB Cloud
+web-based GUI <https://cloud.edgedb.com/>`_.
+
+The first time you access the web UI, you will be prompted to log in.
```
Comment (1st1) [inline docs/guides/cloud.rst:55]: Devon, we're going to be rapidly iterating the cloud UI. Let's drop the screenshots entirely.

File: docs/guides/cloud.rst [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+.. image:: images/cloud/9-data-explorer.png
+    :width: 100%
+    :alt: A screenshot of the EdgeDB Cloud data explorer
+
+
+Find bugs? Have Questions?
+==========================
+
+Thank you for helping us make the best way to host your EdgeDB instances even
+better! Please join us on `our Discord <https://discord.gg/umUueND6ag>`_ to ask
```
Comment (1st1) [inline docs/guides/cloud.rst:124]: or email us at support@edgedb.com

File: docs/guides/cloud.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaclarke) [reply docs/guides/cloud.rst]: https://github.com/edgedb/edgedb.com/pull/622

With that PR this should work:
```
.. raw:: html

    <a class="promoLink" href="/p/cloud-waitlist">
      <span>Join Cloud Waitlist</span>
    </a>
```

File: docs/guides/cloud.rst (hunk 2 above)
Comment (raddevon) [reply docs/guides/cloud.rst:29]: @1st1 Based on my read of @fantix's notes and the test cases in the [PR](https://github.com/edgedb/shared-client-testcases/pull/27) @zackelan referenced, it looks like underscore is also now disallowed. Can someone confirm or deny?

File: docs/guides/cloud.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+graphically, inspect your schema, and browse your data.
+
+
+Find bugs? Have Questions?
+==========================
+
+Thank you for helping us make the best way to host your EdgeDB instances even
+better! Please join us on `our Discord <https://discord.gg/umUueND6ag>`_  or
+email us at `support@edgedb.com <mailto:support@edgedb.com>`_ to ask questions
+and report any bugs you find.
```
Comment (1st1) [inline docs/guides/cloud.rst:100]:
```suggestion
Thank you for helping us make the best way to host your EdgeDB instances even
better! Please join us on `our Discord <https://discord.gg/umUueND6ag>`_ to ask questions. Report any bugs you find at https://edgedb.com/p/cloud-support.
```

(please make it a proper hyperlink; I'm deploying the site now for this URL to become active)

File: docs/cli/edgedb_cloud/index.rst [hunk 7]
```diff
@@ -74,6 +74,18 @@ these commands:
...
+.. note::
+
+    Please be aware of the following restrictions on EdgeDB Cloud instance
+    names:
+
+    * can contain only Latin alpha-numeric characters or ``-``
+    * cannot start with a dash (``-``) or contain double dashes (``--``)
+    * maximum instance name length is 61 characters minus the length of your
+      GitHub username (i.e., length of GitHub username + length of instance
+      name < 62)
```
Comment (1st1) [inline docs/cli/edgedb_cloud/index.rst:86]:
```suggestion
      name must be less than 62 characters.)
```

File: docs/cli/edgedb_instance/edgedb_instance_create.rst [hunk 8]
```diff
@@ -38,6 +38,17 @@ To create a Cloud instance, your instance name should be in the format
...
+.. note::
+
+    Please be aware of the following restrictions on EdgeDB Cloud instance
+    names:
+
+    * can contain only Latin alpha-numeric characters or ``-``
+    * cannot start with a dash (``-``) or contain double dashes (``--``)
+    * maximum instance name length is 61 characters minus the length of your
+      GitHub username (i.e., length of GitHub username + length of instance
+      name < 62)
```
Comment (1st1) [inline docs/cli/edgedb_instance/edgedb_instance_create.rst:50]: ditto

File: docs/cli/edgedb_project/edgedb_project_init.rst [hunk 9]
```diff
@@ -41,6 +41,17 @@ alphanumeric characters and hyphens (i.e., ``-``). You can provide this Cloud
...
+.. note::
+
+    Please be aware of the following restrictions on EdgeDB Cloud instance
+    names:
+
+    * can contain only Latin alpha-numeric characters or ``-``
+    * cannot start with a dash (``-``) or contain double dashes (``--``)
+    * maximum instance name length is 61 characters minus the length of your
+      GitHub username (i.e., length of GitHub username + length of instance
+      name < 62)
```
Comment (1st1) [inline docs/cli/edgedb_project/edgedb_project_init.rst:53]: ditto

**Review by 1st1 — APPROVED** [review]

**Comment by 1st1** [comment]
Let's push this. We'll need another PR to educate people how to use secret keys to connect to cloud instances from vercel et al

**Pull request merged**